src/Main/Character/CharacterParses.js

```diff
@@ -0,0 +1,482 @@
+              ITEMS[singleParse.gear[13].id] = {
+                name: singleParse.gear[13].name,
+                id: singleParse.gear[13].id,
+                icon: ITEMS[0].icon,
+                quality: singleParse.gear[13].quality,
+              };
+            }
+          });
+      });
+    });
+
+    this.fillMissingTrinkets();
+
+    return parses;
+
+  }
+
+  get zoneBosses() {
+    const zones = ZONES.filter(zone => zone.id === this.state.activeZoneID ).map(e => {
+      return e.encounters;
+    })[0];
+    return zones;
+  }
+
+  fetchImage() {
+    //save the chars image URL to localstore
+    const image = localStorage.getItem(`${this.props.region}/${this.props.realm}/${this.props.name}`);
+    if (image) {
+      this.setState({
+        image: image,
+      });
+      return;
+    }
+    
+    return fetch(`https://${this.props.region}.api.battle.net/wow/character/${encodeURIComponent(this.props.realm)}/${encodeURIComponent(this.props.name)}?locale=en_GB&apikey=n6q3eyvqh2v4gz8t893mjjgxsf9kjdgz`)
+      .then(response => response.json())
+      .then((data) => {
+        if (data.status === 'nok') {
+          alert('Character doesn\'t exist');
+          return;
+        }
+        const image = data.thumbnail.replace('-avatar.jpg', '');
+        localStorage.setItem(`${this.props.region}/${this.props.realm}/${this.props.name}`, image);
+        this.setState({
+          image: image,
+        });
+      });
+  }
+
+  load(refresh = false) {
+    this.setState({
+      isLoading: true,
+    });
+
+    let charName = encodeURIComponent(this.props.name);
+    //use the slug from REALMS when available, otherwise try realm-prop and fail
+    let charRealm = REALMS[this.props.region] ? REALMS[this.props.region].realms.filter(elem => { return elem.name === this.props.realm; })[0].slug : this.props.realm;
+    charRealm = encodeURIComponent(charRealm);
+
+    if (process.env.NODE_ENV === 'development') {
+      //temp necessary until WoWA encodes the character-endpoint on its own
+      charName = encodeURIComponent(charName);
+      charRealm = encodeURIComponent(charRealm);
+    }
+    return fetchWcl(`parses/character/${charName}/${charRealm}/${this.props.region}`, {
+      metric: this.state.metric,
+      zone: this.state.activeZoneID,
+      _: refresh ? +new Date() : undefined,
+    }).then((rawParses) => {
+      if (rawParses.status === 400) {
+        // means char was not found on WCL
+        this.setState({
+          error: true,
+        });
+        return;
+      }
+      if (this.state.class !== "") { //only update parses when class was already parsed (since its only a metric/raid change)
+        const parses = this.changeParseStructure(rawParses);
+        this.setState({
+          parses: parses,
+          isLoading: false,
+        });
+
+        return;
+      }
+
+      if (rawParses.length === 0) { //happens when the character has no logs for the selected raid
```

Comment:
Shouldn't this be above the previous if-statement?

Reply:
fixed